controllers/dataplane/openstackdataplanedeployment_controller.go

```diff
@@ -216,7 +216,7 @@ func (r *OpenStackDataPlaneDeploymentReconciler) Reconcile(ctx context.Context,
 					for certKey := range service.Spec.TLSCerts {
 						result, err := deployment.EnsureTLSCerts(ctx, helper, &nodeSet,
 							nodeSet.Status.AllHostnames, nodeSet.Status.AllIPs, service, certKey)
-						if err != nil {
+						if (err != nil) || (result != ctrl.Result{}) {
```

Comment:
Should this use `*result` instead since we do in L233?

Reply:
yes...i think i do. it needs to be dereferenced.